File: Src/java/elm/src/main/java/org/cqframework/cql/elm/evaluating/SimpleElmEvaluator.java
package org.cqframework.cql.elm.evaluating;

import javax.xml.namespace.QName;
import org.hl7.elm.r1.Expression;
import org.hl7.elm.r1.TypeSpecifier;

public class SimpleElmEvaluator {

    private static final SimpleElmEngine engine = new SimpleElmEngine();

    public static SimpleElmEngine simpleElmEngine() {
        return engine;
    }

    public static boolean booleansEqual(Expression left, Expression right) {
        return engine.booleansEqual(left, right);
    }

    public static boolean integersEqual(Expression left, Expression right) {
        return engine.integersEqual(left, right);
    }

    public static boolean decimalsEqual(Expression left, Expression right) {
        return engine.decimalsEqual(left, right);
    }

    public static boolean stringsEqual(Expression left, Expression right) {
        return engine.stringsEqual(left, right);
    }

    public static boolean stringsEqual(String left, String right) {
        return engine.stringsEqual(left, right);
    }

    public static boolean dateTimesEqual(Expression left, Expression right) {
        return engine.dateTimesEqual(left, right);
    }

    public static boolean dateRangesEqual(Expression left, Expression right) {
        return engine.dateRangesEqual(left, right);
    }

    public static boolean codesEqual(Expression left, Expression right) {
        return engine.codesEqual(left, right);
    }

    public static boolean typeSpecifiersEqual(TypeSpecifier left, TypeSpecifier right) {
        return engine.typeSpecifiersEqual(left, right);
    }

    public static boolean qnamesEqual(QName left, QName right) {
        return engine.qnamesEqual(left, right);
    }
}

File: Src/java/elm/src/commonMain/kotlin/org/cqframework/cql/elm/evaluating/SimpleElmEvaluator.kt
package org.cqframework.cql.elm.evaluating

import kotlin.jvm.JvmStatic
import nl.adaptivity.xmlutil.QName
import org.hl7.elm.r1.Expression
import org.hl7.elm.r1.TypeSpecifier

@Suppress("TooManyFunctions")
object SimpleElmEvaluator {
    private val engine = SimpleElmEngine()

    fun simpleElmEngine(): SimpleElmEngine {
        return engine
    }

    fun booleansEqual(left: Expression?, right: Expression?): Boolean {
        return engine.booleansEqual(left, right)
    }

    fun integersEqual(left: Expression?, right: Expression?): Boolean {
        return engine.integersEqual(left, right)
    }

    fun decimalsEqual(left: Expression?, right: Expression?): Boolean {
        return engine.decimalsEqual(left, right)
    }

    @JvmStatic
    fun stringsEqual(left: Expression?, right: Expression?): Boolean {
        return engine.stringsEqual(left, right)
    }

    @JvmStatic
    fun stringsEqual(left: String?, right: String?): Boolean {
        return engine.stringsEqual(left, right)
    }

    fun dateTimesEqual(left: Expression?, right: Expression?): Boolean {
        return engine.dateTimesEqual(left, right)
    }

    @JvmStatic
    fun dateRangesEqual(left: Expression?, right: Expression?): Boolean {
        return engine.dateRangesEqual(left, right)
    }

    @JvmStatic
    fun codesEqual(left: Expression?, right: Expression?): Boolean {
        return engine.codesEqual(left, right)
    }

    @JvmStatic
    fun typeSpecifiersEqual(left: TypeSpecifier?, right: TypeSpecifier?): Boolean {
        return engine.typeSpecifiersEqual(left, right)
    }

    @JvmStatic
    fun qnamesEqual(left: QName?, right: QName?): Boolean {
        return engine.qnamesEqual(left, right)
    }
}
